### ISSUE TITLE
Package bounds prevent developer from upgrading

### ISSUE THREAD

**Issue by timholy** [JuliaLang/Pkg.jl#761]
In developing a breaking version of Interpolations, here's the sequence of events:
- developed it locally
- tested (on julia 1.0) all the dependencies listed in julia 0.6's `Pkg.dependents`, determined which ones would break
- [added upper bounds on the Interpolations version](https://github.com/JuliaLang/METADATA.jl/pull/18101) for those packages
- [tagged a new version (0.9) of Interpolations](https://github.com/JuliaLang/METADATA.jl/pull/18113)
- updated my packages

Now what I want to do is create a new version of ImageTransformations that's compatible with the new version of Interpolations. Unfortunately, I keep getting told I can't have Interpolations 0.9 checked out at the same time as my "dev" copy of ImageTransformations. This is true even though I've done a `activate .` and added a local Project.toml file with the appropriate compatibility for ImageTransformations.

In the end the only way I found to work around it was to edit `~/.julia/registries/General/I/ImageTransformations/Compat.toml` by hand.

**Comment by timholy** [comment]
Hah, I think I must have `free`d ImageTransformations and forgotten to `dev` it again.

I'll close this. Sorry this is mostly noise. But perhaps worth a small comment that this is not the first time I've done this to myself. Not that I have any ideas about what to do about it.

**Comment by timholy** [comment]
Maybe when we print messages like this:
```
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package CoordinateTransformations [150eb455]:
 CoordinateTransformations [150eb455] log:
 ├─possible versions are: [0.1.0, 0.2.0, 0.3.0-0.3.2, 0.4.0-0.4.1, 0.5.0] or uninstalled
 ├─restricted to versions 0.4.0-0.4 by ImageTransformations [02fcd773], leaving only versions 0.4.0-0.4.1
 │ └─ImageTransformations [02fcd773] log:
 │   ├─possible versions are: 0.5.2 or uninstalled
 │   └─ImageTransformations [02fcd773] is fixed to version 0.5.2
 └─restricted to versions 0.5.0 by an explicit requirement — no versions left
```
we should indicate the free/dev status of each involved package? (Note the message above was generated as I was experimenting with adding Project.toml file to ImageTransformations, and has nothing to do with the error above.)
